graylog2-web-interface/src/views/logic/queries/Query.ts

```diff
@@ -149,14 +149,14 @@ export default class Query {
   // eslint-disable-next-line @typescript-eslint/no-use-before-define
   toBuilder(): Builder {
     const { id, query, timerange, filter, searchTypes } = this._value;
+
     // eslint-disable-next-line @typescript-eslint/no-use-before-define
-    const builder = Query.builder()
+    return Query.builder()
       .id(id)
       .query(query)
       .timerange(timerange)
-      .searchTypes(searchTypes);
-
-    return filter ? builder.filter(filter) : builder;
+      .searchTypes(searchTypes)
+      .filter(filter);
```

Comment:
I am not completely sure why we implemented this filter specific logic previously.

Reply:
The idea was to set the `filter` attribute only if it was specified, leaving the default otherwise.

Reply:
Do you think we should keep the check?
Another solution could be implementing `return filter !== undefined ? builder.filter(filter) : builder;`. But as far as I can tell the result will be the same.

Reply:
I updated the implementation as discussed.